### PR TITLE
Modularize Microbenchmark Infrastructure

### PR DESCRIPTION
**Pull request by ncteisen**
This takes the old bm_diff, and splits it into modules that can be used separately for faster iteration (you no longed have to rebuild and run your baseline comparison branch). The PR should currently give the exact same behavior as before.

### CODE REVIEW

File: tools/profiling/microbenchmarks/bm_build.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+### Python utility to build opt and counters benchmarks """
+
+import bm_constants
```
Comment (ctiller) [inline tools/profiling/microbenchmarks/bm_build.py]: It's probably worth having tools/run_tests/run_microbenchmark.py use this module also.

**Comment by grpc-testing** [comment]
```
No significant performance differences
```

**Comment by ncteisen** [comment]
Mac test failures #11223, Linux test failures #11077 (which will be fixed in #11384).

**Comment by ncteisen** [comment]
Switching reviewers since Craig is going on vacation soon. Feel free to skip this review and skip straight to #11367, which is built on this

**Comment by ncteisen** [comment]
Closed upstream by #11367

**Pull request closed without merging**